docs: Kubernetes on Azure with CoreOS and Weave

errordeveloper opened the pull request:
In this tutorial we will demonstrate how to deploy a Kubernetes cluster to Azure cloud. Weave makes networking of containers simple and secure, in a transparent, yet robust way. The focus of this tutorial is to provide an out-of-the-box production-ready implementation with dedicated Kubernetes master and etcd nodes. It will also show how to scale the cluster with ease.

errordeveloper committed:
Import errordeveloper/weave-demos@44f542f97bae857729200bbeca1969929f5fb9f3

errordeveloper commented on docs/getting-started-guides/coreos/azure/cloud_config_templates/kubernetes-cluster-main-nodes-template.yml:
```diff
@@ -0,0 +1,351 @@
+        Documentation=https://github.com/GoogleCloudPlatform/kubernetes
+        Requires=network-online.target
+        [Service]
+        Environment=KUBE_RELEASE_TARBALL=https://github.com/GoogleCloudPlatform/kubernetes/releases/download/v0.9.3/kubernetes.tar.gz
```
I am using _v0.9.3_ due to #4414.

errordeveloper commented:
Pinging @kelseyhightower.

erictune commented:
@errordeveloper please take a look at 
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/README.md
and tell me if you think this merits a separate line item from the existing Azure support,  and if you want to be listed as the point of contact.

errordeveloper commented:
@erictune I didn't realise there is that file, I do think it needs to be updated also. This was done in collaboration with Microsoft team - @squillace and @chanezon, who probably can be listed also. Unless you'd wish one contact there, otherwise we will have to decide who it shall be.

erictune commented:
multiple contacts is fine.

chanezon commented:
@erictune cc @jeffmendoza yes I think it deserves a separate line. Jeff's docs refers to using salt to deploy, ubuntu and openvpn for networking. This one is about Weave for networking, on CoreOS, using cloud-init for deployment.

erictune commented:
feel free to add such a line in this PR.

errordeveloper commented:
@erictune see 80450b9. I've included a networking column there as well, as now there are more options.

errordeveloper committed:
docs: Kubernetes on Azure with CoreOS and Weave -

- improve the wording
- suggest next step
- remove unrelated code
- minor fix to instructions
- checkin empty dir

errordeveloper committed:
docs: Update the index of guides -

- add entry for CoreOS on Azure with Weave
- add networking column

erictune commented:
Thanks for updating the README.  The new column is fine.

errordeveloper commented:
@erictune I was wondering if the reviewer would be required to test this kind of pull-request manually at this point..?

chanezon commented:
Works, http://kubernetes-service-cluster-43e4ff223570ba.cloudapp.net/
with a few issues to fix, and additional docs needed.

1- script should create the output directory. If it doesn't exists yet, it stops.
2- right now when provisioning the cloud service, the script tries to create kubernetesservicecluster account owned by someone else, so cloud service creation fails
Only way I found to bypass that is creating the storage account in same region, and specify credentials via env variables.
Would be better to create a storage account dynamically.
export AZURE_STORAGE_ACCOUNT=patkube
export AZURE_STORAGE_ACCESS_KEY="71Py+BzdVEI...FmDBKOcydSY3FT+0gCToxT9bFvyCJZXjg=="
at least add that to docs.
3- Typo in instructions for guestbook deployment 
kubectl create -f redis-master-controller.json
not redis-master.json
4- kubectl create -f frontend-service.json
F0224 21:38:17.284492    2407 create.go:78] the cloud provider does not support external TCP load balancers.

You need to remove createExternalLoadBalancer and add publicIPs with internal Azure IPs for minions.

adding the Azure internal ips for the minion machines to the service json file, then creating an endpoint to these on the same port does the trick.
Creating one endpoint for each minion on different ports let me verify that I could hit both.
I ended up creating something that would look more like a prod scenario, with endpoints for the VM instances which publicIPs you mentioned in the service manifest, belonging to a single load balanced set. That provides you 2 levels of load balancing.
I guess all that could be automated.
{
  "id": "frontend",
  "kind": "Service",
  "apiVersion": "v1beta1",
  "port": 8000,
  "publicIPs": [
        "172.18.0.14",
        "172.18.0.13"
    ],
  "containerPort": "http-server",
  "selector": {
    "name": "frontend"
  },
  "labels": {
    "name": "frontend"
  },
 
}

jeffmendoza commented:
The ubuntu/salt Azure deployment uses (openvpn)[https://openvpn.net/] for networking.

erictune commented:
@errordeveloper  I think at this point, it isn't practical for reviewers to test these types of PRs, so we are trusting that you have tested it.  

Since Kelsey hasn't responded in a while, I'll assume he is busy and pick up this review.

Looks like you need a rebase, so lets start there.

chanezon commented:
I've tested it and it works, if @errordeveloper fixes the 3 issues I mention in the comment above https://github.com/GoogleCloudPlatform/kubernetes/pull/4483#issuecomment-75893512

errordeveloper commented:
Ok I am back on thins now. Here is a list based on @chanezon's comment:

  - [x] fix output directory issue (eee9ffb1ca5423e1f734940f130ab4f09cbfd93b)
  - [ ] automate creation of storage account (or at least document the workaround)
  - [ ] Typo in instructions for guestbook deployment (`kubectl create -f redis-master-controller.json` not `redis-master.json`)
  - [ ] fix `kubectl create -f frontend-service.json` results in `the cloud provider does not support external TCP load balancer`

errordeveloper committed:
Create Azure storage account if needed

errordeveloper commented:
>  However, I will need to test the latest release and see if the fix for #4414 had been merged into it as wells as if the changes to Guestbook PHP example has appeared there too.

I can see there is 0.11.0 release candidate, it's probably best to wait for the release itself...

errordeveloper commented:
I've now tested 0.11.0, and looks like it works, and, in fact, it's a no longer a release candidate :+1:

I'll pushing an update once confirm it's definitely working well.

errordeveloper committed:
coreos/azure: Version bump to 0.11.0

errordeveloper committed:
coreos/azure: Ensure docs match the output in new version

errordeveloper commented:
The issue I am seeing with Guestbook PHP example is this:
```
Uncaught exception 'Predis\Connection\ConnectionException' with message 'php_network_getaddresses: getaddrinfo failed: Name or service not known [tcp://:0]' in /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php:141
```
...which is [documented](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/examples/guestbook/README.md), and indeed the `REDIS_MASTER_SERVICE_HOST` and `REDIS_MASTER_PORT` are not set, according to `docker inspect`.

Any ideas what could be wrong with this and where should I look?

erictune commented:
The reason that the env vars are not set is that you are passing the `etcd_servers` option to kubelet.  That option no longer supported and we are in the process of removing it.  
You need to specify `--api_servers`.  Take a look at what #4708 did with kubelet.

erictune commented:
sorry, accidental push of close button.

erictune commented:
You can do this two ways.
You can have the API server export its service on its private IP, port 8080, without authentication.  This is fine if you trust everyone on your cluster's private network.  Use the firewall to protect 8080 from outsiders and only poke a hole in the firewall to the nginx proxy.  For this you have to set `--address=0.0.0.0` on the apiserver.  And you tell the kubelet to talk to `--api_servers=http://$MASTERIP:8080`

Or, you can only expose an authorized port on the internal network.  For this you set `--address=127.0.01 --auth_port=6443` on the apiserver.  And you tell the kubelet to talk to `--api_servers=https://$MASTERIP:6443`.  But you have to generate a kubelet token and pass that to the master and the kubelets.
  Happy to elaborate.

errordeveloper commented:
@erictune thank you, I'll push a fix now. I tried changing proxy arguments too, but it looks like it still wants etcd servers... would be good to know if changes to proxy are expected soon or it's to stay the way it is?

errordeveloper committed:
coreos/azure: Point kublet and proxy at API server instead of etcd

errordeveloper committed:
coreos/azure: Remove workaround coreos/bugs#267 -

it turned out the issue was due to the domain lenght, so let's just
keep it short.

erictune commented:
Oh, right.  The kube-proxy flagname is `--master=` while the kubelet flagname is `--api_servers`.  Argument should be the same.

errordeveloper commented:
@erictune I'll update proxy's arguments and retest tomorrow, also it'd be good if @chanezon or @squillace could test it once again before merging anyway.

errordeveloper commented:
@erictune I've updated the arguments passed to proxy and rebased. This is ready to merge, as far as I am concerned, unless Microsoft folks and/or @kelseyhightower have any input.

erictune commented on docs/getting-started-guides/coreos/azure/README.md:
````diff
@@ -0,0 +1,187 @@
+
+To get started, you need to checkout the code:
+```
+git clone https://github.com/GoogleCloudPlatform/kubernetes
````
can you change this to explicitly pull the v0.11.0 tag?  That way, if upstream changes, your guide will keep working until you next update and test i.t

errordeveloper replied:
Just thought this would a good idea at first, but then realised that won't be able to pull-request into a tag... I could put the exact revision, but that's also impossible since revision is unknown until I commit it. In fact, this guide is pointing user to the JavaScript code that is part of the guide and Go code is not compiled as part of this or anything like that. I am considering it to be safe to keep the user on master here, as I am not expecting anyone to touch this Azure provisioning code much at all.

erictune commented:
One minor request, and then good to merge.

erictune commented:
LGTM.

chanezon commented:
Awesome to see this merged: as of today, this Weave/CoreOS tool and doc is the only way I was able to provision a Kubernetes cluster in Azure.

I have not tested the Ubuntu/OpenVPN way but heard there were some issues with it in current Kubernetes version.

I have tested the Flannel docs and after https://github.com/GoogleCloudPlatform/kubernetes/pull/4362 was fixed I ran into https://github.com/GoogleCloudPlatform/kubernetes/issues/4141 and stopped there. It may work, but needs further testing and documentation.

rajnemani commented:
Hi all,
Trying to deploy Kubernetes on Azure/CoreOS/Weave.  I cannot hit my Azure virtual IP from out side for the Guestbook app.  I can login to master.  Run 'kubectl get services'.  
Here is my services output
NAME           LABELS                                    SELECTOR            IP(S)          PORT(S)
frontend       name=frontend                             name=frontend       10.1.243.130   80/TCP
kubernetes     component=apiserver,provider=kubernetes   <none>              10.1.0.1       443/TCP
redis-master   name=redis-master                         name=redis-master   10.1.190.67    6379/TCP
redis-slave    name=redis-slave                          name=redis-slave    10.1.140.115   6379/TCP

Curl to frontend gives valid HTML output
Curl to apiserver gives blank output which is probably ok
Curl to redis-master gives invalid number of arguments for 'get' which is proably ok
Curl to  redis-slave is same as master above

The cluster is running on one master, 4 kube nodes and 3 etcd machines (one master and 2 nodes? I am not sure on etcd). 

This is deployed to Azure.  Once the deployment fineshes all the 8 machines have the same Virtual IP and different private IPs (consistent with the subnet range they are in).  I added the 8000 port endpoint on kube-01.  But cannot hit the VIP.  What could I be doing wrong.

Thanks for any help
Raj

errordeveloper commented:
Are you using the master branch?

On Sat, 22 Aug 2015 16:54 rajnemani <notifications@github.com> wrote:

> Hi all,
> Trying to deploy Kubernetes on Azure/CoreOS/Weave. I cannot hit my Azure
> virtual IP from out side for the Guestbook app. I can login to master. Run
> 'kubectl get services'.
>
> Here is my services output
> NAME LABELS SELECTOR IP(S) PORT(S)
> frontend name=frontend name=frontend 10.1.243.130 80/TCP
> kubernetes component=apiserver,provider=kubernetes 10.1.0.1 443/TCP
> redis-master name=redis-master name=redis-master 10.1.190.67 6379/TCP
> redis-slave name=redis-slave name=redis-slave 10.1.140.115 6379/TCP
>
> Curl to frontend gives valid HTML output
> Curl to apiserver gives blank output which is probably ok
> Curl to redis-master gives invalid number of arguments for 'get' which is
> proably ok
> Curl to redis-slave is same as master above
>
> The cluster is running on one master, 4 kube nodes and 3 etcd machines
> (one master and 2 nodes? I am not sure on etcd).
>
> This is deployed to Azure. Once the deployment fineshes all the 8 machines
> have the same Virtual IP and different private IPs (consistent with the
> subnet range they are in). I added the 8000 port endpoint on kube-01. But
> cannot hit the VIP. What could I be doing wrong.
>
> Thanks for any help
> Raj
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-133717414>
> .
>

rajnemani commented:
Hi Ilya,
Thank you for your time.

I am using the instructions at
https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/getting-started-guides/coreos/azure/README.md

to set up Kubernetes on Azure/CoreOs/Weave.  Per the instructions there I
am cloning from

https://github.com/GoogleCloudPlatform/kubernetes.

Actually after a bit of investigation and help, I can get the
Guestbook app to load.  The documentation incorrectly states that the
endpoint port

mapping in AZURE is 8000 (public) to 8000 (private).  But the private
port turned out to be one in the range for nodePort

type services (30000-32767).

But I have a new problem now.  The frontend PHP application cannot
talk to redis backend.  After a bit of research I

figured that it needs DNS setup.  So started looking at

https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md

Here I got couple of questions. Based on the information in the DNS
setup doc above, I modified the cluster/azure/config-default.sh

with the following.

ENABLE_CLUSTER_DNS="${KUBE_ENABLE_CLUSTER_DNS:-true}"
DNS_SERVER_IP="10.0.0.10"
DNS_DOMAIN="cluster.local"
DNS_REPLICAS=1


Then I ran the https://github.com/kubernetes/kubernetes/tree/master/docs/getting-started-guides/coreos/azure/create-kubernetes-cluster.js
<https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/coreos/azure/create-kubernetes-cluster.js>

which created the cluster on Azure.  Then I ran the DNS server
replication controller and Service from

https://github.com/kubernetes/kubernetes/tree/master/docs/getting-started-guides/coreos/azure/addons

on the Kubernetes master node.  I ran the validation test to see if my
DNS is working by following the steps in the DNS setup

document.  Here is the output of the test

core@kube-00 ~/addons $ kubectl exec busybox -- nslookup kubernetes
Server:    10.1.0.3
Address 1: 10.1.0.3

Name:      kubernetes
Address 1: 10.1.0.1

The IPs above do not correspond to the DNS Server IP I setup at
cluster/azure/config-default.sh.  It looks like these IPs are are
coming from DNS replication controllers and Service
YAML files.

So my questions are

1. What is significance of the DNS entries in the
cluster/azure/config-default.sh file.  Are they used at all in the

overall setup?

2.  The DNS document also talks about starting the Kubectl with DNS
server IP and DNS Domain parameters.  How do I do

that.  Also which 'set' of IP and Domain values do I use for starting
Kubectl, the ones from cluster/azure/config-default.sh file or the
ones

mentioned in the DNS replication controller and service yaml files
(Cluster IP: 10.1.0.3, domain:kube.local).


I know this is a long email but wanted to provide details so you can
follow what I am asking.

Can you provide some guidance?

Is there an updated document that has all these steps outlined.

Thanks in advance for your time and help.

Raj





On Mon, Aug 24, 2015 at 1:36 AM, Ilya Dmitrichenko <notifications@github.com
> wrote:

> Are you using the master branch?
>
> On Sat, 22 Aug 2015 16:54 rajnemani <notifications@github.com> wrote:
>
> > Hi all,
> > Trying to deploy Kubernetes on Azure/CoreOS/Weave. I cannot hit my Azure
> > virtual IP from out side for the Guestbook app. I can login to master.
> Run
> > 'kubectl get services'.
> >
> > Here is my services output
> > NAME LABELS SELECTOR IP(S) PORT(S)
> > frontend name=frontend name=frontend 10.1.243.130 80/TCP
> > kubernetes component=apiserver,provider=kubernetes 10.1.0.1 443/TCP
> > redis-master name=redis-master name=redis-master 10.1.190.67 6379/TCP
> > redis-slave name=redis-slave name=redis-slave 10.1.140.115 6379/TCP
> >
> > Curl to frontend gives valid HTML output
> > Curl to apiserver gives blank output which is probably ok
> > Curl to redis-master gives invalid number of arguments for 'get' which is
> > proably ok
> > Curl to redis-slave is same as master above
> >
> > The cluster is running on one master, 4 kube nodes and 3 etcd machines
> > (one master and 2 nodes? I am not sure on etcd).
> >
> > This is deployed to Azure. Once the deployment fineshes all the 8
> machines
> > have the same Virtual IP and different private IPs (consistent with the
> > subnet range they are in). I added the 8000 port endpoint on kube-01. But
> > cannot hit the VIP. What could I be doing wrong.
> >
> > Thanks for any help
> > Raj
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-133717414>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-134096483>
> .
>

errordeveloper commented:
@rajnemani please look in [master](https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/coreos/azure/README.md#kubernetes-on-azure-with-coreos-and-weave) instead, and also see https://github.com/kubernetes/kubernetes/issues/12224#issuecomment-132158579 and https://github.com/kubernetes/kubernetes/issues/12907.

rajnemani commented:
Ok, I can see the updated doc with information on how to get the frontend port configuration.  Can I ask if working from master (cloning from master) is also going to resolve the DNS issues between frontend and redis backend i.e I do not have to do all the DNS stuff manually?

Thanks
Raj

errordeveloper commented:
> Can I ask if working from master (cloning from master) is also going to resolve the DNS issues between frontend and redis backend i.e I do not have to do all the DNS stuff manually?

Yes.

rajnemani commented:
Hi,
Thank you for your help so far.  After cloning from master, do I need to
run "make release' command to build the source or can I start running the
cluster setup script assuming pre-built binaries?
Thanks
Raj


On Mon, Aug 24, 2015 at 9:36 AM, Ilya Dmitrichenko <notifications@github.com
> wrote:

> Can I ask if working from master (cloning from master) is also going to
> resolve the DNS issues between frontend and redis backend i.e I do not have
> to do all the DNS stuff manually?
>
> Yes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-134292445>
> .
>

errordeveloper commented:
> do I need to run "make release' command to build the source or can I start running the cluster setup script assuming pre-built binaries?

No, please just follow the instruction [as given](https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/coreos/azure/README.md#lets-go), the provisioning scripts use released binaries and you don't need to build anything.

rajnemani commented:
Hi Ilya,

I can get the app up but cannot post message to redis.  It is the same
issue that I encountered before.  Here is the error.  I am not sure the
instructions are complete.  You can see it cannot find
tcp:redis-slave:6379.  It could be that I need to setup DNS manually after
all?
Thank you for your help
-------------------

Failed to load resource: the server responded with a status of 404 (Not
Found)
angular.min.js:85 TypeError: Cannot read property 'push' of undefined
    at RedisController.onRedis (controllers.js:9)
    at angular.min.js:163
    at angular.min.js:179
    at h.$eval (angular.min.js:102)
    at h.$apply (angular.min.js:103)
    at HTMLButtonElement.<anonymous> (angular.min.js:179)
    at angular.min.js:27
    at q (angular.min.js:7)
    at HTMLButtonElement.Yc.c (angular.min.js:27)(anonymous function) @
angular.min.js:85
angular.js:9435 TypeError: Cannot read property 'push' of undefined
    at RedisController.onRedis (controllers.js:9)
    at angular.js:10220
    at angular.js:18003
    at h.$eval (angular.js:11977)
    at h.$apply (angular.js:12077)
    at HTMLButtonElement.<anonymous> (angular.js:18002)
    at angular.js:2616
    at q (angular.js:310)
    at HTMLButtonElement.Yc.c (angular.js:2615)(anonymous function) @
angular.js:9435(anonymous function) @ angular.js:6846h.$apply @
angular.js:12079(anonymous function) @ angular.js:18002(anonymous function)
@ angular.js:2616q @ angular.js:310Yc.c @ angular.js:2615
controllers.js:26 <br />
<b>Fatal error</b>:  Uncaught exception
'Predis\Connection\ConnectionException' with message
'php_network_getaddresses: getaddrinfo failed: Name or service not known
[tcp://redis-slave:6379]' in
/vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php:141
Stack trace:
#0 /vendor/predis/predis/lib/Predis/Connection/StreamConnection.php(96):
Predis\Connection\AbstractConnection-&gt;onConnectionError('php_network_get...',
0)
#1 /vendor/predis/predis/lib/Predis/Connection/StreamConnection.php(70):
Predis\Connection\StreamConnection-&gt;tcpStreamInitializer(Object(Predis\Connection\ConnectionParameters))
#2 /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php(96):
Predis\Connection\StreamConnection-&gt;createResource()
#3 /vendor/predis/predis/lib/Predis/Connection/StreamConnection.php(144):
Predis\Connection\AbstractConnection-&gt;connect()
#4 /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php(181):
Predis\Connection\StreamConnection-&gt;connect()
#5 /vendor/predis/predis/lib/Predis/Connection/StreamConnectio in
<b>/vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php</b>
on line <b>141</b><br />

angular.js:9435 TypeError: Cannot read property 'split' of undefined
    at controllers.js:27
    at angular.js:7576
    at A (angular.js:10965)
    at A (angular.js:10965)
    at angular.js:11051
    at h.$eval (angular.js:11977)
    at h.$digest (angular.js:11803)
    at h.$apply (angular.js:12083)
    at f (angular.js:7841)
    at H (angular.js:8042)(anonymous function) @ angular.js:9435(anonymous
function) @ angular.js:6846A @ angular.js:10968A @
angular.js:10965(anonymous function) @ angular.js:11051h.$eval @
angular.js:11977h.$digest @ angular.js:11803h.$apply @ angular.js:12083f @
angular.js:7841H @ angular.js:8042C.onreadystatechange @ angular.js:7985

On Mon, Aug 24, 2015 at 1:03 PM, Ilya Dmitrichenko <notifications@github.com
> wrote:

> do I need to run "make release' command to build the source or can I start
> running the cluster setup script assuming pre-built binaries?
>
> No, please just follow the instruction as given
> <https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/coreos/azure/README.md#lets-go>,
> the provisioning scripts use released binaries and you don't need to build
> anything.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-134359746>
> .
>

errordeveloper commented:
Are you sure the DNS server is running and accessible?

On Tue, 25 Aug 2015 00:05 rajnemani <notifications@github.com> wrote:

> Hi Ilya,
>
> I can get the app up but cannot post message to redis. It is the same
> issue that I encountered before. Here is the error. I am not sure the
> instructions are complete. You can see it cannot find
> tcp:redis-slave:6379. It could be that I need to setup DNS manually after
> all?
> Thank you for your help
> -------------------
>
> Failed to load resource: the server responded with a status of 404 (Not
> Found)
> angular.min.js:85 TypeError: Cannot read property 'push' of undefined
> at RedisController.onRedis (controllers.js:9)
> at angular.min.js:163
> at angular.min.js:179
> at h.$eval (angular.min.js:102)
> at h.$apply (angular.min.js:103)
> at HTMLButtonElement.<anonymous> (angular.min.js:179)
> at angular.min.js:27
> at q (angular.min.js:7)
> at HTMLButtonElement.Yc.c (angular.min.js:27)(anonymous function) @
> angular.min.js:85
> angular.js:9435 TypeError: Cannot read property 'push' of undefined
> at RedisController.onRedis (controllers.js:9)
> at angular.js:10220
> at angular.js:18003
> at h.$eval (angular.js:11977)
> at h.$apply (angular.js:12077)
> at HTMLButtonElement.<anonymous> (angular.js:18002)
> at angular.js:2616
> at q (angular.js:310)
> at HTMLButtonElement.Yc.c (angular.js:2615)(anonymous function) @
> angular.js:9435(anonymous function) @ angular.js:6846h.$apply @
> angular.js:12079(anonymous function) @ angular.js:18002(anonymous function)
> @ angular.js:2616q @ angular.js:310Yc.c @ angular.js:2615
> controllers.js:26 <br />
> <b>Fatal error</b>: Uncaught exception
> 'Predis\Connection\ConnectionException' with message
> 'php_network_getaddresses: getaddrinfo failed: Name or service not known
> [tcp://redis-slave:6379]' in
> /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php:141
> Stack trace:
> #0 /vendor/predis/predis/lib/Predis/Connection/StreamConnection.php(96):
>
> Predis\Connection\AbstractConnection-&gt;onConnectionError('php_network_get...',
> 0)
> #1 /vendor/predis/predis/lib/Predis/Connection/StreamConnection.php(70):
>
> Predis\Connection\StreamConnection-&gt;tcpStreamInitializer(Object(Predis\Connection\ConnectionParameters))
> #2 /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php(96):
> Predis\Connection\StreamConnection-&gt;createResource()
> #3 /vendor/predis/predis/lib/Predis/Connection/StreamConnection.php(144):
> Predis\Connection\AbstractConnection-&gt;connect()
> #4 /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php(181):
> Predis\Connection\StreamConnection-&gt;connect()
> #5 /vendor/predis/predis/lib/Predis/Connection/StreamConnectio in
> <b>/vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php</b>
> on line <b>141</b><br />
>
> angular.js:9435 TypeError: Cannot read property 'split' of undefined
> at controllers.js:27
> at angular.js:7576
> at A (angular.js:10965)
> at A (angular.js:10965)
> at angular.js:11051
> at h.$eval (angular.js:11977)
> at h.$digest (angular.js:11803)
> at h.$apply (angular.js:12083)
> at f (angular.js:7841)
> at H (angular.js:8042)(anonymous function) @ angular.js:9435(anonymous
> function) @ angular.js:6846A @ angular.js:10968A @
> angular.js:10965(anonymous function) @ angular.js:11051h.$eval @
> angular.js:11977h.$digest @ angular.js:11803h.$apply @ angular.js:12083f @
> angular.js:7841H @ angular.js:8042C.onreadystatechange @ angular.js:7985
>
> On Mon, Aug 24, 2015 at 1:03 PM, Ilya Dmitrichenko <
> notifications@github.com
> > wrote:
>
> > do I need to run "make release' command to build the source or can I
> start
> > running the cluster setup script assuming pre-built binaries?
> >
> > No, please just follow the instruction as given
> > <
> https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/coreos/azure/README.md#lets-go
> >,
> > the provisioning scripts use released binaries and you don't need to
> build
> > anything.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-134359746>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/4483#issuecomment-134406964>
> .
>
